### PR TITLE
[Flight Reply] Encode FormData

### PR DESCRIPTION
**Pull request by sebmarkbage**
Builds on top of https://github.com/facebook/react/pull/26661

This lets you pass FormData objects through the Flight Reply serialization. It does that by prefixing each entry with the ID of the reference and then the decoding side creates a new FormData object containing only those fields (without the prefix).

Ideally this should be more generic. E.g. you should be able to pass Blobs, Streams and Typed Arrays by reference inside plain objects too. You should also be able to send Blobs and FormData in the regular Flight serialization too so that they can go both directions. They should be symmetrical. We'll get around to adding more of those features in the Flight protocol as we go.

### CODE REVIEW

File: packages/react-client/src/ReactFlightReplyClient.js [hunk 1]
```diff
@@ -185,6 +191,24 @@ export function processReply(
         );
         return serializePromiseID(promiseId);
       }
+      // TODO: Should we the Object.prototype.toString.call() to test for cross-real objects?
```
Comment (sophiebits) [inline packages/react-client/src/ReactFlightReplyClient.js:194]:
```suggestion
      // TODO: Should we the Object.prototype.toString.call() to test for cross-realm objects?
```

File: packages/react-server/src/ReactFlightReplyServer.js [hunk 2]
```diff
@@ -397,6 +409,23 @@ function parseModelString(
...
+        const formPrefix = response._prefix + stringId + '_';
+        const data = new FormData();
+        const backingFormData = response._formData;
+        // We assume that the reference to FormData always comes after each
+        // entry that it references so we can assume they all exist in the
+        // backing store already.
+        // $FlowFixMe[prop-missing] FormData has forEach on it.
+        backingFormData.forEach((entry: File | string, entryKey: string) => {
+          if (entryKey.startsWith(formPrefix)) {
+            data.append(entryKey.substr(formPrefix.length), entry);
```
Comment (sophiebits) [inline packages/react-server/src/ReactFlightReplyServer.js:424]: substr is kinda deprecated? annex B.

```suggestion
            data.append(entryKey.substring(formPrefix.length), entry);
```

(personally I'm a slice gal)

**Review by sophiebits — APPROVED** [review]
I didn't review the #26661 part

File: packages/react-server/src/ReactFlightReplyServer.js (hunk 2 above)
Comment (sebmarkbage) [reply packages/react-server/src/ReactFlightReplyServer.js:424]: I'd be down to just lint and enforce a single form just for the potential compression savings by using a repeated string.

**Comment by sebmarkbage** [comment]
Oops. I forgot that I had a stacked PR. Anyway. Probably fine.

**Pull request merged**